### ISSUE TITLE
Implement filters parameters for /books API endpoint

### ISSUE THREAD

**Issue by richard015ar** [pressbooks/pressbooks#3312]
As a developer I would like to get a filtered list of books from `<NETWORK_URL>/wp-json/pressbooks/v2/books` endpoint. These filters should allow me to specify certain values and operators for certain properties.

Example: retrieve books with `bookDirectoryExcluded` false, with licenses code different than `All Rights Reserved` and with `wordCount` metadata greater than 5000.

The parameters I'd like to use are:
- bookDirectoryExcluded [boolean]
- license code [string]
- name [string]
- wordCount [integer]

These filters must be inclusive and must accept different operators like: `!= / not contains, >/greater than, ==/equals, etc`.

Filters we should support are the following:
- T/F for booleans
- contains/does not contain for strings
- `>=`/`<=` for integers

DoD: extend/modify the current books API endpoint to accept and process these filters.

**Comment by richard015ar** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pressbooks-Product-Backlog-60ae7d807f6e370010b3d5be/issues/pressbooks/pressbooks/3312?planning-poker) with Zenhub @arzola @fdalcin

**Comment by richard015ar** [comment]
Update: I got a working version at https://github.com/pressbooks/pressbooks/tree/feat/extend-books-endpoint.

Right now in the books endpoint, we get the [paginated books](https://github.com/pressbooks/pressbooks/blob/9ea8cb18b7d11877062b4f2605f9d6de418ace57/inc/api/endpoints/controller/class-books.php#L331) ids from the `wp_blogs` table, and then, [we read most of the metadata](https://github.com/pressbooks/pressbooks/blob/9ea8cb18b7d11877062b4f2605f9d6de418ace57/inc/api/endpoints/controller/class-books.php#L257) from `wp_blogmeta`.

In order to filter books what I did as a POC (testing code) is to join the `wp_blogmeta` table, where title, license code, words count, etc are.
The testing code I did, works well.
The problem is that this data is from the data collector, populated by network analytics, which is not available for open source, so this solution won't work as it is.

So I looked at the source of these metadata, and it is in `wp_<BLOG>_posts` for `post_type = metadata `. In the data collector, we get it using the getBookInformation method, which switches the blog.

So, basically, these parameters won't make much sense for open source users, in fact, open source users shouldn't see much metadata in the book endpoint, except for the `bookDirectoryExcluded` field, which is differently handled.

I'm planning to meet with the tech team to discuss it, and also notify Steel before moving forward with this.

**Comment by richard015ar** [comment]
Update:

Blogmeta is populated as expected for open-source users through some hooks when updating the site and posts.

After a quick chat with Dalcin to discuss the implementation, I got a working solution at https://github.com/pressbooks/pressbooks/tree/feat/extend-books-endpoint.

The only pending portion is feature tests. I'll work on them tomorrow.

**Comment by SteelWagstaff** [comment]
We need to create a new ticket for excluding licenses. Otherwise this ticket can be closed as done.
